api/Program.cs

```diff
@@ -16,8 +15,23 @@
 {
   options.SerializerOptions.TypeInfoResolverChain.Insert(0, AppJsonSerializerContext.Default);
 });
+
 builder.Services.AddValidation();
 
+// With this, validation errors will be returned as a ProblemDetails response.
+builder.Services.AddProblemDetails(options =>
+{
+    options.CustomizeProblemDetails = (context) =>
+    {
+        if (context.ProblemDetails is HttpValidationProblemDetails validationProblem)
+        {
+            context.ProblemDetails.Detail = $"Error(s) occurred: {validationProblem.Errors.Values.Sum(x => x.Length)}";
+        }
+
+        context.ProblemDetails.Extensions.TryAdd("timestamp", DateTime.UtcNow.ToString("o", CultureInfo.InvariantCulture));

```

Comment:
Could I trouble you to add some code here that showcases how this can be used to set the property keys on the errors object to comply with the JSON casing policy defined in serializer options?

It won't be 100% foolproof but should hopefully provide a way for folks to address this.